### ISSUE TITLE
Replace the `|` operator with `dj.OrList`

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-matlab#77]
The `|` operator is currently used to construct or-lists for restrictions but it is limited to restrictions by relations.  In `datajoint-python`, we implement the explicit `dj.OrList` object.  I propose to match the matlab implementation and deprecate the `|` operator.

**Comment by eywalker** [comment]
What would be a reason not to support either syntax? Use of `|` sounds intuitive compared to having to use `dj.OrList`.

**Comment by dimitri-yatsenko** [comment]
The `|` operator only works for objects of classes that we define and is therefore limited.  `dj.OrList` works for any types of restrictors.  If `dj.OrList` works, then i would rather have one correct way of doing something.

**Comment by eywalker** [comment]
Do we support something like restricting by a cell array?

**Comment by dimitri-yatsenko** [comment]
Yes, an `OrList` is any list.  So what we really need is an `AndList` so that
```matlab
r & a & b & c
```
can be expressed as 
```matlab
r & dj.AndList({a b c})
```

**Comment by ixcat** [comment]
should be implemented with #96
